WcaOnRails/spec/controllers/competitions_controller_spec.rb

```diff
@@ -660,4 +660,64 @@
       end
     end
   end
+
+  describe 'GET #edit_events' do
+    context 'when not signed in' do
+      sign_out
+
+      it 'redirects to the sign in page' do
+        get :edit_events, id: competition.id
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+
+    context 'when signed in as an admin' do
+      sign_in { FactoryGirl.create :admin }
+
+      it 'shows the edit competition events form' do
+        get :edit_events, id: competition.id
+        expect(response).to render_template :edit_events
+      end
+    end
+
+    context 'when signed in as a regular user' do
+      sign_in { FactoryGirl.create :user }
+
+      it 'does not allow access' do
+        expect {
+          get :edit_events, id: competition.id
+        }.to raise_error(ActionController::RoutingError)
```

Comment:
I agree, I think that's more consistent with how we do stuff elsewhere. In fact, it looks like you're using `redirect_unless_user`, so I'm surprised we'd get a `RoutingError` here.